File: sparkSqlProcess/src/main/java/com/neunn/sparkSqlProcess/MySqlAction.java
package com.neunn.sparkSqlProcess;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MySqlAction {
	private static SimpleDateFormat ft = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss,S");
	
	private String host;
	private String port;
	private String databaseName;
	private String userName;
	private String passWord;
	
	public MySqlAction(String host, String port, String databaseName, String userName, String passWord){
		this.host = host;
		this.port = port;
		this.databaseName = databaseName;
		this.userName = userName;
		this.passWord = passWord;
	}
	
	public Connection getConnection(){
		Connection con = null; //定义一个MYSQL链接对象
        try {
			Class.forName("com.mysql.jdbc.Driver").newInstance();
			con = DriverManager.getConnection("jdbc:mysql://" + host + ":" + port + "/" + databaseName, userName, passWord); //链接本地MYSQL
        }catch (Exception e) {
			// TODO: handle exception
        	System.out.println(e);
		}
        return con;
	}
	
	public void update(String sqlString){
		Statement stmt; //创建声明
        try {
        	System.out.println(ft.format(new Date()) + " sqlString is: " + sqlString);
			stmt = getConnection().createStatement();
			//stmt.executeUpdate("UPDATE spark_sql_histrory_query set save_flag = 'success' where auto_id = 34");
			stmt.executeUpdate(sqlString);
		} catch (SQLException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

    /**
     * 入口函数
     * @param arg
     */
    public static void main(String arg[]) {
    	new MySqlAction("10.2.1.10", "3306", "db_nbds", "nbds_user", "neunbds").update("");
    }
}

File: sparkSqlCrestTable/src/main/scala/com/neunn/MySqlAction.scala
package com.neunn

import java.sql.DriverManager
import java.sql.Connection
import java.text.SimpleDateFormat
import java.util.Date

import scala.collection.mutable
import scala.collection.mutable.ArrayBuffer

/**
  * Created by xuzq on 16-8-9.
  */
class MySqlAction(val host:String, val port:String, val dbName:String, val userName:String, val passwd:String) {

  classOf[com.mysql.jdbc.Driver]
  val ft = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss,S")

  def getConnection(host:String, port:String, dbName:String, userName:String, passwd:String): Connection = {
    val connection = DriverManager.getConnection("JDBC:mysql://%s:%s/%s".format(host, port, dbName), userName, passwd)
    connection
  }

  def close(conn: Connection): Unit ={
    try{
      if(!conn.isClosed() || conn != null){
        conn.close()
      }
    }catch {
      case ex: Exception =>{
        ex.printStackTrace()
      }
    }
  }

  def query(sqlString: String): ArrayBuffer[mutable.HashMap[String, Any]]={
    println(ft.format(new Date()) + " the sqlString is: " + sqlString)
    val conn = getConnection(host, port, dbName, userName, passwd)

    try{
      val pstm = conn.prepareStatement(sqlString)

      val rs = pstm.executeQuery()

      val rsmd = rs.getMetaData()
      val size = rsmd.getColumnCount()
      val buffer = new ArrayBuffer[mutable.HashMap[String, Any]]()

      while(rs.next()){
        val map = mutable.HashMap[String, Any]()
        for (i <- 1 to size){
          map += (rsmd.getColumnLabel(i) -> rs.getString(i))
        }
        buffer.append(map)
      }
      buffer
    }
    finally {
      conn.close()
    }
  }

  def modify(sqlString: String): Unit ={
    println(ft.format(new Date()) + " the sqlString is: " + sqlString)
    val conn = getConnection(host, port, dbName, userName, passwd)
    try{
      val pstm = conn.prepareStatement(sqlString)
      pstm.executeUpdate()
    }catch {
      case ex: Exception => {
        ex.printStackTrace()
      }
    }finally{
      conn.close()
    }
  }

  def add(sqlString :String): Unit ={
    println(ft.format(new Date()) + " the sqlString is: " + sqlString)
    val conn = getConnection(host, port, dbName, userName, passwd)
    try{
      val pstm = conn.prepareStatement(sqlString)
      pstm.executeUpdate()
    }catch {
      case ex: Exception => {
        ex.printStackTrace()
      }
    }finally {
      conn.close()
    }
  }
}
